File: my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::chrono_literals;

class NumberPublisher : public rclcpp::Node
{
public:
    NumberPublisher(): Node("number_publisher")
    {
        this->declare_parameter("number", 2);
        this->declare_parameter("timer_period", 1.0);

        number_ = this->get_parameter("number").as_int();
        double timer_period = this->get_parameter("timer_period").as_double();

        publisher_ = create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = create_timer(std::chrono::duration<double>(timer_period), std::bind(&NumberPublisher::NumberPublish, this));
        RCLCPP_INFO(get_logger(), "Numbers started");
    }
private:
    void NumberPublish()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        RCLCPP_INFO(get_logger(), "Publishing: %ld", msg.data);
        publisher_->publish(msg);
    }
    int number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/number_publisher.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class NumberPublisher(Node): 
    def __init__(self):
        super().__init__("number_publisher")
        
        self.declare_parameter("number", 2)
        self.declare_parameter("timer_period", 1.0)

        self.number_ = self.get_parameter("number").value
        self.timer_period_ = self.get_parameter("timer_period").value
        self.get_logger().info(f"Number: {self.number_}, Timer_period: {self.timer_period_}")
        
        self.publisher_= self.create_publisher(Int64, "number", 10)
        self.timer = self.create_timer(1.0, self.publish_number)
        self.get_logger().info("Number Publisher has started")
    
    def publish_number(self):
        msg = Int64()
        msg.data = self.number_
        self.publisher_.publish(msg)
        self.get_logger().info(f"published: {msg.data}")

def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisher()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()